### DISCUSSION TITLE
Question regarding the calculations of velocity confidence

### DISCUSSION THREAD

**Discussion by paula-tataru** [theislab/scvelo#702] [Q&A]
Hi,

I was looking into how the velocity confidence is calculated and I noticed that the mean velocity is subtracted twice. The first step is to remove the mean per cell across all velocity genes. 
```python
V -= V.mean(1)[:, None]
```

Then, when the confidence for a cell is calculated, the mean per neighbor is again subtracted from the already adjusted velocities.
```python
Vi_neighs = V[indices[i]]
Vi_neighs -= Vi_neighs.mean(1)[:, None]
```

What is the reason for subtracting the mean twice in the neighbors' velocities?

**Comment by WeilerP** [comment]
@VolkerBergen, any thoughts on this?
